src/lib/layers/editable-polygons-layer.js

```diff
@@ -1,148 +1,268 @@
-// @flow
-import { point, polygon, multiPolygon } from '@turf/helpers';
+/* eslint-env browser */
 
-import LayerMouseEvent from '../layer-mouse-event';
-import Feature from '../feature';
+import { CompositeLayer, GeoJsonLayer, ScatterplotLayer } from 'deck.gl';
+import { immutablyReplaceCoordinate } from '../geojson';
 
-import PolygonsLayer from './polygons-layer';
-import EditableJunctionsLayer from './editable-junctions-layer';
+export default class EditablePolygonsLayer extends CompositeLayer {
```

Comment:
How feasible/meaningful do you think it would be to do something like:

```
class EditableLayer extends CompositeLayer {
  // Event handling registration functions, ...
}

class EditablePolygonLayer extends EditableLayer {
  //...
}
```

Reply:
absolutely! once i implement another editable layer, i'll factor out shared functionality. great idea